oemoflex/preprocessing.py

```diff
@@ -0,0 +1,261 @@
+
+        # Set up the skeleton of the output dataframe consisting of attribute names as
+        # column titles and default values
+        component_data = {
+            c_attr['attribute']: c_attr['default'] for _, c_attr in component_attrs.iterrows()
+        }
+
+        component_suffix = {
+            c_attr['attribute']: c_attr['suffix'] for _, c_attr in component_attrs.iterrows()
+        }
+
+        # Fill 'region' with country code list
+        if component == 'link':
+            # Generate region column of the form "AT_DE"
+            component_data['region'] = [code.replace('-', '_') for code in link_list]
+
+            # Reserve 'name' column because there is no suffix to use here
+            # line could be dropped by defining a suffix such as '-link'
+            component_data['name'] = [code for code in link_list]
+
+            # for the two bus attributes reserve the colums with a part of the country code
+            component_data['from_bus'] = [code.split('-')[0] for code in link_list]
+            component_data['to_bus'] = [code.split('-')[1] for code in link_list]
+
+        else:
+            component_data['region'] = country_list
+
+        # Fill other columns with their respective suffixes if available
+        for attr_name, suffix in component_suffix.items():
+
+            # If a suffix has to be applied
+            if not pd.isna(suffix):
+
+                # for 'link' element use the pre-defined name part instead of the region
+                if attr_name == 'from_bus' or attr_name == 'to_bus':
+                    component_data[attr_name] = [link + suffix for link in component_data[attr_name]]
+
+                else:
+                    component_data[attr_name] = [code + suffix for code in component_data['region']]
+
+        df = pd.DataFrame(component_data).set_index('region')
+
+        # Write to target directory
+        df.to_csv(os.path.join(dir, component + '.csv'))
+
+
+def get_parameter_values(scalars_df, parameter_name):
+    r"""
+    Selects rows from common input file "Scalars.csv" by column=='parameter_name'
+    and maintains the relation 'Region' -> 'Value' at external assignment
+
+    Parameters
+    ----------
+    scalars_df : DataFrame
+    DataFrame of "Scalars.csv"
+
+    parameter_name : str
+    Specifies the rows to select by the name in column "Parameter"
+
+    Returns
+    -------
+    The parameter's values (column 'Value') as a DataFrame, indexed by 'Region'
+    """
+
+    return scalars_df.loc[scalars_df['Parameter'] == parameter_name].set_index('Region')['Value']
+
+
+def update_shortage_file(data_preprocessed_path):
+    logging.info("Updating shortage file")
+
+    shortage_file = os.path.join(data_preprocessed_path, 'elements', 'shortage.csv')
+
+    # Read prepared CSV file
+    shortage = pd.read_csv(shortage_file, index_col='region')
+
+    # Fill column 'marginal_cost' with a fixed value for ALL the elements
+    shortage['marginal_cost'] = 5000
```

Comment:
This should be read from the input data. Doing so requires to pass scalars to the function as it is done with the others.

Reply:
Only possible with Scalars.csv v0.02 - should we do this in another PR addressing changes for switching to v0.02?

Reply:
Alternatively, I would do a final commit to this PR updating to v0.02. OK?

Reply:
Ok, let's do this in #18